### DISCUSSION TITLE
Japanese language file

### DISCUSSION THREAD

**Discussion by coolvitto** [Timthreetwelve/GetMyIP#115] [General]
Hello,

This is the Japanese language file. 
Please add it.

[Strings.ja-JP.zip](https://github.com/user-attachments/files/17892491/Strings.ja-JP.zip)

- - - - - - - -
Best regards,

  coolvitto

**Comment by Timthreetwelve** [comment]
@coolvitto

Hello. Thank you so much for contributing your translation. It was such a pleasant surprise to see it in my inbox this morning.

It seems that I have added one more string to the en-US file after your translation. If you don't mind and have the time, would you translate the following?

`<sys:String x:Key="MsgText_ImportSettingsRestart">Please restart Get My IP to use the imported settings.</sys:String>`

If it is more convenient, you can paste the translated string in this discussion.

There is also an opportunity to add Japanese to the installer if you would like. The file that contains the translations is [here](GetMyIP/Inno_Setup/GetMyIPLocalization.iss). It would need one line added in the `[Messages]` section and then a block of 6 lines added to the `[CustomMessages]` section.

Once again, you can paste the translated strings in this discussion, or attach the entire file, whichever is more convenient.

Thank you so very much,

Tim

**Comment by Timthreetwelve** [comment]
Screenshots 
![GetMyIP_2024-11-24_14-48-35](https://github.com/user-attachments/assets/0c649100-c7b0-4e85-b663-7ef0c98fd867)

![GetMyIP_2024-11-24_14-30-01](https://github.com/user-attachments/assets/ae25d8de-5f84-4cef-b495-5249e9a15c0c)

**Comment by coolvitto** [comment]
<sys:String x:Key="MsgText_ImportSettingsRestart">インポートした設定を使用するには、Get My IP を再起動してください。</sys:String>

please

**Comment by coolvitto** [comment]
>" It would need one line added in the [Messages] section and then a block of 6 lines added to the [CustomMessages] section."

I don't quite understand what this means. 
Please provide some examples in English.

**Comment by Timthreetwelve** [comment]
@coolvitto 

My projects use Inno Setup to create the installers (exe files). Inno Setup provides a way to use multiple languages in those installers. Messages can be changed, or custom messages can be defined. Below you will find the messages that need to be translated.

The message in the [Messages] section basically just needs to have the work "Setup" translated, the text in the curly braces is substituted at runtime.

The messages in the [CustomMessages ] section need the string to the right of the equals sign (=) translated. The %n signifies a new line and should remain as they are. The %1 and %2 are placeholders for program name and version and should remain but can be moved so that the phrase is grammatically correct.

When translating the messages, replace the leading "en." with "ja.", keep the message identifier (e.g. DeleteConfigFiles=), then translate everything to the right of the equals sign (=).

You can paste the translated items in a comment if that is easier.

```
[Messages]
en.SetupWindowTitle = Setup - {#MyAppName} {#MyAppVersion}

[CustomMessages]
en.AppIsRunning=is running, please close it to continue with the installation.
en.DeleteConfigFiles=Do you want to remove the settings files?%n%nSelect 'No' if you plan on reinstalling.
en.NotSelfContained=This will install the standard version of %1 version %2.%n%nThis version requires an existing installation of .NET 8 Desktop Runtime and is compatible with both x64 and x86 systems.%n%nIt Is recommended that you close all other applications before continuing.%n%nClick 'Next' to continue, or 'Cancel' to exit Setup.
en.SelfContainedx64=This will install the self-contained x64 (64-bit) version of %1 version %2.%n%nIt Is recommended that you close all other applications before continuing.%n%nClick 'Next' to continue, or 'Cancel' to exit Setup.
en.SelfContainedx86=This will install the self-contained x86 (32-bit) version of %1 version %2.%n%nIt Is recommended that you close all other applications before continuing.%n%nClick 'Next' to continue, or 'Cancel' to exit Setup.
en.ViewReadme=View the ReadMe file
```

Thanks,
Tim

**Comment by coolvitto** [comment]
Am I correct in understanding that the above message is not necessary for the portable version?

**Comment by Timthreetwelve** [comment]
No, it's just for the installer versions. 

If you're not interested in contributing to the installer  versions, I understand. I appreciate everything you've done. 

Thanks,
Tim

**Comment by coolvitto** [comment]
you're welcome.
